modvendor/modvendor.go

```diff
@@ -66,35 +67,42 @@ type modinfo struct {
 // reference them inside the vendor directory.
 //
 // It returns a report of everything vendored and ignored (with a reason).
-func Vendor(rootdir string, vendorDir string, modsrc tf.Source) Report {
+func Vendor(rootdir string, vendorDir project.Path, modsrc tf.Source) Report {
 	report := NewReport(vendorDir)
-	if !filepath.IsAbs(vendorDir) {
+	if !path.IsAbs(vendorDir.String()) {
 		report.Error = errors.E("vendor dir %q must be absolute path", vendorDir)
 		return report
 	}
-
 	return vendor(rootdir, vendorDir, modsrc, report, nil)
 }
 
 // VendorAll will vendor all dependencies of the tfdir into rootdir.
 // It will scan all .tf files in the directory and vendor each module declaration
 // containing the supported remote source URLs.
-func VendorAll(rootdir string, vendorDir string, tfdir string) Report {
+func VendorAll(rootdir string, vendorDir project.Path, tfdir string) Report {
 	return vendorAll(rootdir, vendorDir, tfdir, NewReport(vendorDir))
 }
 
-// Dir returns the directory for the vendored module source, relative to project
+// TargetDir returns the directory for the vendored module source, relative to project
 // root.
-func Dir(vendorDir string, modsrc tf.Source) string {
-	return filepath.Join(vendorDir, modsrc.Path, modsrc.Ref)
+//
+// On Windows, when modsrc.Scheme is "file" it replaces the volume “:“ by `$` because
+// `:` is disallowed as path component in such system.
```

Comment:
Just a remainder to update this according to our previous discussions :100:

Reply:
(discussed in slack already)